hypothesis-python/src/hypothesis/strategies/_internal/types.py

```diff
@@ -552,6 +563,74 @@ def resolve_Iterator(thing):
     return st.iterables(st.from_type(thing.__args__[0]))
 
 
+@register(typing.Counter, st.builds(collections.Counter))
+def resolve_Counter(thing):
+    return st.dictionaries(
+        keys=st.from_type(thing.__args__[0]),
+        values=st.integers(),
+    ).map(collections.Counter)
+
+
+@register(typing.Deque, st.builds(collections.deque))
+def resolve_deque(thing):
+    return st.lists(st.from_type(thing.__args__[0])).map(collections.deque)
+
+
+@register(typing.ChainMap, st.builds(dict).map(collections.ChainMap))
+def resolve_ChainMap(thing):
+    return resolve_Dict(thing).map(collections.ChainMap)
+
+
+@register("OrderedDict", st.builds(dict).map(collections.OrderedDict))
+def resolve_OrderedDict(thing):
+    # typing.OrderedDict is new in Python 3.7.2
+    return resolve_Dict(thing).map(collections.OrderedDict)
+
+
+@register(typing.Pattern, st.builds(re.compile, st.sampled_from(["", b""])))
+def resolve_Pattern(thing):
+    if isinstance(thing.__args__[0], typing.TypeVar):
+        return st.builds(re.compile, st.sampled_from(["", b""]))
+    return st.just(re.compile(thing.__args__[0]()))
+
+
+@register(
+    typing.Match,
+    st.text().map(lambda c: re.match(".", c, flags=re.DOTALL)).filter(bool),
+)
+def resolve_Match(thing):
+    if thing.__args__[0] == bytes:
+        return (
+            st.binary(min_size=1)
+            .map(lambda c: re.match(b".", c, flags=re.DOTALL))
+            .filter(bool)
+        )
+    return st.text().map(lambda c: re.match(".", c, flags=re.DOTALL)).filter(bool)
+
+
+class GeneratorStrategy(st.SearchStrategy):
+    def __init__(self, yields, returns):
+        assert isinstance(yields, st.SearchStrategy)
+        assert isinstance(returns, st.SearchStrategy)
+        self.yields = yields
+        self.returns = returns
+
+    def __repr__(self):
+        return f"<generators yields={self.yields!r} returns={self.returns!r}>"
+
+    def do_draw(self, data):
+        elements = conjecture_utils_many(data, min_size=0, max_size=100, average_size=5)
+        while elements.more():
+            yield data.draw(self.yields)
+        return data.draw(self.returns)
+
+
+@register(typing.Generator, GeneratorStrategy(st.none(), st.none()))
+def resolve_Generator(thing):
+    yields, _, returns = thing.__args__
```

Comment:
Are we able to send something to this generator?

Reply:
As per your comment below, yes - it's a real generator, so we can `.send(...)` to it.  

However we don't do anything with the sent value, as runtime type-checking is not possible in general (e.g. consuming an `Iterable` is destructive, many generics don't support `isinstance`, etc), and so the send-type annotation is ignored.